src/handlers/handleDirectMessage.test.ts

```
@@ -0,0 +1,235 @@
+        await handleDirectMessage(ctx, 'admin-group-123');
+
+        expect(ctx.db.saveMessage).toHaveBeenCalledTimes(1);
+        expect(ctx.db.saveMessage).toHaveBeenCalledWith(expect.objectContaining({ id: '123', type: 'user' }));
+
+        const saveMessageOrder = vi.mocked(ctx.db.saveMessage).mock.invocationCallOrder[0];
+        const forwardMessageOrder = vi.mocked(ctx.api.forwardMessage).mock.invocationCallOrder[0];
+
+        expect(saveMessageOrder).toBeLessThan(forwardMessageOrder);
+    });
+
+    it('should log errors when forward message fails', async () => {
```

ESLint

[error] Test has no assertions

(vitest/expect-expect)